src/initialization.jl

```diff
@@ -29,7 +29,14 @@ function __init__()
     # register device overrides
     precompiling = ccall(:jl_generating_output, Cint, ()) != 0
     if !precompiling
-        eval(overrides)
+        eval(Expr(:block, overrides))
+        empty!(overrides)
+
+        @require SpecialFunctions="276daf66-3868-5448-9aa4-cd146d93841b" begin
```

Comment:
Urgh... Losing the ability to define compat bounds makes me strongly dislike Requires...

Reply:
Yeah. In this case, we could actually keep SpecialFunctions as a package dependency, but just not import it to avoid regressing load time. On the other hand, SpecialFunctions seems pretty stable, so let's do that when we run into incompatibilities?

Reply:
Ah we still have it as a dependency! Great, then I am okay with this.

Reply:
No, we could. Actually, no reason we shouldn't. Let's add that back.